mofacts/server/methods.js

```diff
@@ -713,6 +713,21 @@ async function setExperimentState(UserId, TDFId, newExperimentState) { // by cur
   return TDFId;
 }
 
+async function insertHiddenItem(userId, clusterKC) {
+  let query = 'UPDATE componentstate SET showitem = FALSE WHERE userid = $1 AND kcid = $2';
+  await db.manyOrNone(query, [userId, clusterKC]);
+}
+
+async function getHiddenItems(userId, tdfId) {
+  let query = 'SELECT kcid FROM componentstate WHERE userid = $1 AND tdfid = $2 AND showitem = false';
+  const hiddenItems = await db.manyOrNone(query, [userId, tdfId]);
+  let ret = [];
+  for(let i = 0; i < hiddenItems.length; i++){
+    ret.push(hiddenItems[i].kcid);
+  }
+  return [...new Set(ret)];
```

Comment:
@MegaGeese Seems like you deleted the deduplication here in the new version?

Reply:
Yes. Since the new query specifications limit the returned hiddenItems to a unique set of stimulus KC IDs. So there would be no point in going from a set to an array.